### PR TITLE
Option to log queries that last more than x seconds

### PR DESCRIPTION
**Pull request by skalero01**
Added a config file and an option to select the min_size of the query to be logged.
Added a option to disable the package too.

This way the package can be used on production mode if that is desired.

Regards! Hope this request helps the package to be better.

### CODE REVIEW

File: src/config/laravel-query-logger.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+return [
+
+	'enabled' => true,
+
+    /*
+     * Min size of query to log in seconds
+     */
+
+    'min_size' => 0,
```
Comment (overtrue) [inline src/config/laravel-query-logger.php]: I think the name `min_size` is not clearable.

File: src/config/laravel-query-logger.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+return [
+
+	'enabled' => true,
```
Comment (overtrue) [inline src/config/laravel-query-logger.php]: `'enabled' => env('QUERY_LOGGER_ENABLED', true),`  ?

File: src/ServiceProvider.php [hunk 3]
```diff
@@ -23,16 +23,26 @@ class ServiceProvider extends LaravelServiceProvider
...
-
             $bindings = $query->connection->prepareBindings($query->bindings);
             $pdo = $query->connection->getPdo();
             $realSql = vsprintf($sqlWithPlaceholders, array_map([$pdo, 'quote'], $bindings));
-            $duration = $this->formatDuration($query->time / 1000);
-
-            Log::debug(sprintf('[%s] %s', $duration, $realSql));
+            $seconds = $query->time / 1000;
+            $duration = $this->formatDuration($seconds);
+            if($seconds >= config('laravel-query-logger.min_size', 0)) {
```
Comment (overtrue) [inline src/ServiceProvider.php]: Plz follow PSR-2 code standard https://www.php-fig.org/psr/psr-2/

File: src/config/laravel-query-logger.php (hunk 1 above)
Comment (overtrue) [reply src/config/laravel-query-logger.php]: Maybe milliseconds is more useful?

File: src/ServiceProvider.php [hunk 4]
```diff
@@ -23,16 +23,26 @@ class ServiceProvider extends LaravelServiceProvider
      */
     public function boot()
     {
-        Log::info(sprintf('============ %s: %s ===============', request()->method(), request()->fullUrl()));
+        // Config files
+        $this->publishes([realpath(__DIR__) . '/config/laravel-query-logger.php' => config_path('laravel-query-logger.php')], 'config');
```
Comment (overtrue) [inline src/ServiceProvider.php]: I think the config filename should be `laravel-query-logger.php`?

File: src/config/laravel-query-logger.php (hunk 1 above)
Comment (skalero01) [reply src/config/laravel-query-logger.php]: What name do you think its better? maybe `minimum_size_in_miliseconds`?

File: src/ServiceProvider.php (hunk 3 above)
Comment (skalero01) [reply src/ServiceProvider.php]: Not sure what you mean here, Put the the config value on a variable?

File: src/config/query-logger.php [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+return [
+
+	'enabled' => env('QUERY_LOGGER_ENABLED', true),
+
+    /*
+     * Min size of query to log in miliseconds
+     */
+
+    'miliseconds' => 0,
```
Comment (overtrue) [inline src/config/query-logger.php]: `long_query_time`

File: src/ServiceProvider.php [hunk 6]
```diff
@@ -23,16 +23,29 @@ class ServiceProvider extends LaravelServiceProvider
...
             $pdo = $query->connection->getPdo();
             $realSql = vsprintf($sqlWithPlaceholders, array_map([$pdo, 'quote'], $bindings));
-            $duration = $this->formatDuration($query->time / 1000);
-
-            Log::debug(sprintf('[%s] %s', $duration, $realSql));
+            $max_miliseconds = config('query-logger.miliseconds', 0);
+            $seconds = $query->time / 1000;
+            $duration = $this->formatDuration($seconds);
+            $miliseconds = $seconds * 1000;
+            if($miliseconds >= $max_miliseconds) {
```
Comment (overtrue) [inline src/ServiceProvider.php]:
```diff
- if($miliseconds >= $max_miliseconds) {
+ if ($miliseconds >= $max_miliseconds) {
```

**Comment by justmd5** [comment]
这个能pull request 能merge了发新版本么，线上环境需要灵活配置下SQL日志的开关，感谢

**Comment by overtrue** [comment]
@justmd5 你可以用 MySQL 的慢日志解决这个问题吧，感觉不想把这个包弄复杂了，只是一个开发工具。另外可以考虑用 https://github.com/laravel/telescope 来做更灵活的处理。

**Pull request closed without merging**